tests/foreman/cli/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,629 @@
+        assert module_azurerm_cr.name == result_info['compute-attributes'][0]['compute-resource']
+        assert AZURERM_RG_DEFAULT in result_info['compute-attributes'][0]['vm-attributes']
+        assert AZURERM_VM_SIZE_DEFAULT in result_info['compute-attributes'][0]['vm-attributes']
+        assert username in result_info['compute-attributes'][0]['vm-attributes']
+        assert password in result_info['compute-attributes'][0]['vm-attributes']
+        assert AZURERM_PLATFORM_DEFAULT in result_info['compute-attributes'][0]['vm-attributes']
+        assert script_command in result_info['compute-attributes'][0]['vm-attributes']
+        assert AZURERM_FILE_URI in result_info['compute-attributes'][0]['vm-attributes']
+
+
+@run_in_one_thread
+class TestAzureRm_FinishTemplate_Provisioning:
+    """ AzureRM Host Provisioning Tests
+
+    """
+
+    @pytest.fixture(scope='class', autouse=True)
+    def class_setup(self, request, module_domain, module_azurerm_cr, module_azurerm_finishimg):
+        """
+        Sets Constants for all the Tests, fixtures which will be later used for assertions
+        """
+        request.cls.region = settings.azurerm.azure_region
+        request.cls.rhel7_ft_img = AZURERM_RHEL7_FT_IMG_URN
+        request.cls.rg_default = AZURERM_RG_DEFAULT
+        request.cls.premium_os_disk = AZURERM_PREMIUM_OS_Disk
+        request.cls.platform = AZURERM_PLATFORM_DEFAULT
+        request.cls.vm_size = AZURERM_VM_SIZE_DEFAULT
+        request.cls.hostname = gen_string('alpha')
+        request.cls.fullhostname = '{}.{}'.format(self.hostname, module_domain.name).lower()
+
+        request.cls.compute_attrs = (
+            "resource_group={},vm_size={},username={},ssh_key_data={},"
+            "platform={},script_command={},script_uris={},premium_os_disk={}".format(
+                self.rg_default,
+                self.vm_size,
+                module_azurerm_finishimg.username,
+                settings.azurerm.ssh_pub_key,
+                self.platform,
+                'touch /var/tmp/test.txt',
+                AZURERM_FILE_URI,
+                self.premium_os_disk,
+            )
+        )
+        nw_id = module_azurerm_cr.available_networks()['results'][-1]['id']
+        request.cls.interfaces_attributes = (
+            "compute_network={},compute_public_ip=Static," "compute_private_ip=false".format(nw_id)
+        )
+
+    @pytest.fixture(scope='class')
+    def class_host_ft(
+        self,
+        azurermclient,
+        module_azurerm_finishimg,
+        module_azurerm_cr,
+        module_architecture,
+        module_domain,
+        module_location,
+        module_org,
+        module_os,
+        module_smart_proxy,
+        module_puppet_environment,
+    ):
+        """
+        Provisions the host on AzureRM using Finish template
+        Later in tests this host will be used to perform assertions
+        """
+        hammer_api_timeout_set()
```

Comment:
Setting notimeout `-1` to avoid hammer error `Timed out reading data from server` as host create/delete takes time (depends on Azure Cloud response)